File: PalindromeChecker.py
#Function to check string is palindrome or not in python!

# Here we will find reverse of the string and then Check if reverse and original are same or not.
# the predefined function ‘ ‘.join(reversed(string)) is used to reverse string.

def isPalindrome(s):

	rev = ''.join(reversed(s))

	# Checking if both string ar equal or not
	if (s == rev):
		return True
	return False

# main function

s = input("Enter your String: ")
ans = isPalindrome(s)

if (ans):
	print("Yes")
else:
	print("No")

File: PalindromeChecker.cpp
#include<bits/stdc++.h>
using namespace std;

//palindrome checking using two pointers in C++;

bool isPalindrome(string s){
    int i=0;
    int j=s.size()-1;
    while(i<=j){
        if(s[i]!=s[j]) return false;

        i++;
        j--;
    }
    return true;
}
int main(){
  string s;cin>>s;
  if(isPalindrome(s)) cout<<"YES , given string is palindrome"<<endl;
  else cout<<"NO, it is not a palindrome"<<endl;
}
